dev/tasks/README.md

```diff
@@ -0,0 +1,208 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# Arrow Packaging
+
+The content of this directory aims for automating the historically error prone
+process of Arrow packaging.
+
+Packages:
+- C++ and Python [conda-forge packages](conda-recipes) for Linux, Mac and
+  Windows
+- Python [Wheels](python-wheels) for Linux, Mac and Windows
+- C++ and GLib [Linux packages](linux-packages) for multiple distributions
+- Java
+
+## Architecture
+
+### Executors
+
+Individual jobs are executed on public CI services, currently:
+- Linux: TravisCI
+- Mac: TravisCI
+- Windows: AppVeyor
+
+### Queue
+
+Because of the nature of how the CI services work, the scheduling of jobs happens
+through an additional git repository, which acts like a job queue for the tasks.
+A job is a git commit on a particular git branch, containing only the required
+configuration file to run the requested build (currently `.travis.yml` or
+`appveyor.yml`).
+
+### Scheduler
+
+[Crossbow.py](crossbow.py) handles version generation, task rendering and
+submission. The tasks are defined in `tasks.yml`
+
+
+## Install
+
+> The following guide depends on GitHub, but theoretically any git server can be
+> used.
+
+1. [Create the queue
+   repository](https://help.github.com/articles/creating-a-new-repository)
+2. Enable [TravisCI](https://travis-ci.org/getting_started) and
+   [Appveyor](https://www.appveyor.com/docs/) integrations on it
+3. Clone the newly created, by default the scripts looks for `crossbow` next to
+   arrow repository.
+
+   ```bash
+   git clone https://github.com/<user>/crossbow crossbow
+   ```
+
+4. [Create a Personal Access
+   Token](https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/)
+5. Export the token as an environment variable:
+
+   ```bash
+   export CROSSBOW_GITHUB_TOKEN=<token>
+   ```
+
+   > or pass as an argument to the CLI script `--github-token`
+
+6. Install the python dependencies for the script:
+
+   ```bash
+   conda install -y jinja2 pygit2 click pyyaml
+   ```
+
+   ```bash
+   # pygit2 requires libgit2: http://www.pygit2.org/install.html
+   pip install -y jinja2 pygit2 click pyyaml
+   ```
+
+7. Try running it:
+   ```bash
+   $ python crossbow.py --help
+   ```
+
+
+## Usage
+
+The script does the following:
+
+1. Detects the current repository, thus supports forks. The following snippet
+   will build kszucs's fork instead of the upstream apache/arrow repository.
+
+   ```bash
+   $ git clone https://github.com/kszucs/arrow
+   $ git clone https://github.com/kszucs/crossbow
+
+   $ cd arrow/dev/tasks
+   $ python crossbow.py
+   ```
+
+2. Gets the HEAD commit of the currently checked out branch and generates
+   the version number based on [setuptools_scm](https://pypi.python.org/pypi/setuptools_scm).
```

Comment:
BTW We can self generate the version via [pygit2.Repository.describe()](http://www.pygit2.org/repository.html#pygit2.Repository.describe), but then how would we use the same version scheme as `setuptools_cm` in `setup.py`?

Reply:
Using 

```
import setuptools_scm.git
describe = setuptools_scm.git.DEFAULT_DESCRIBE + " --match 'apache-arrow-[0-9]*'"
```

we get a git describe command that output the correct version. The most important part is the additional match, otherwise this is just plain describe to the latest tag.